plugins/bean-validation/src/main/java/org/apache/struts/beanvalidation/validation/interceptor/BeanValidationInterceptor.java

```diff
@@ -95,36 +97,42 @@ protected String doIntercept(ActionInvocation invocation) throws Exception {
         if (LOG.isDebugEnabled()) {
             LOG.debug("Validating [{}/{}] with method [{}]", invocation.getProxy().getNamespace(), invocation.getProxy().getActionName(), methodName);
         }
+        Class<?>[] validationGroup = getValidationGroups(action, methodName);
```

Comment:
Fixed
got a good hint  on..LOG.isDebugEnabled()...
https://logging.apache.org/log4j/2.0/manual/api.html 'Substituting Parameters'

Reply:
Oh! I'm sorry :( I did not mean those logging lines. I meant line#100, `Class<?>[] validationGroup = getValidationGroups(action, methodName);` This line is better to being moved inside that if, please. i.e. when user is specified a skip validation annotation, there is no need to check for validationGroup annotation :) logging lines were ok and should being reverted, thanks.

Reply:
facepalm :-)

Reply:
Please do not delete ` if (LOG.isDebugEnabled()) {`. It seems they are there because of performance when computing the string to log is expensive :) thanks!

Reply:
> got a good hint on..LOG.isDebugEnabled()...
> https://logging.apache.org/log4j/2.0/manual/api.html 'Substituting Parameters'

👍 you're right. Thanks! Please ignore my previous comment. I found some time to read above your mentioned link and you were ok to delete them 😉

Reply:
The ```if (LOG.isDebugEnabled()) {``` practice is not longer needed in all cases when using placeholders like found in the log statement. Preliminary String concatenation is avoided by this new pattern.
Nevertheless, all given parameters will be evaluated before the actual method call, especially ```invocation.getProxy().getNamespace(), invocation.getProxy().getActionName()``` - if these two calls are costly, the if-guard is still justified!

Reply:
👍 sorry... I failed to catch this because of some workload that day :( Thanks!

Reply:
No worries :)